libia2/CMakeLists.txt

```diff
@@ -3,6 +3,12 @@ project(libia2)
 
 add_library(libia2 ia2.c threads.c main.c exit.c)
 target_compile_options(libia2 PRIVATE "-fPIC")
+if(LIBIA2_MTE)
+    target_compile_definitions(libia2 PUBLIC IA2_USE_MTE=1)
+    target_compile_options(libia2 PUBLIC "-march=armv8.5-a+memtag" "-ffixed-x18")
```

Comment:
We should take a closer look at the behavior of `-ffixed-x18`. In my armtest repo, even when compiling with that flag, X18 still seems to get saved and restored between function calls. System libraries also frequently alter x18, so we'll have to compile the whole shebang from libc up with this flag as well.